src/path.ts

```diff
@@ -86,6 +87,7 @@ export function parseRelativeUrl(url: string): {pathname: string; search: string
 }
 
 export function within(root: string, path: string): boolean {
+  const {relative, normalize, resolve, isAbsolute} = op;
   path = relative(normalize(resolve(root)), normalize(resolve(path)));
```

Comment:
The main issue was that `process.cwd()` returns a “namespaced” absolute path starting with _e.g._ `D:\`, whereas `resolve(path)` returns a non-namespaced absolute path. See these logs for reference:

https://github.com/observablehq/framework/actions/runs/11717337869/job/32636834499